### DISCUSSION TITLE
Participate in both oneshot and std

### DISCUSSION THREAD

**Discussion by snippet2021** [yasserfarouk/scml#100] [Q&A] [answered]
I have been creating agents based on OneShotAgent. And I submitted this program to OneShot League.
Now I also would like to be in a Std league with a similar negotiation strategy to my OneShotAnget. Is this possible? I see two possibilities in my mind.

(1) To be in two leagues as two different agents.
Is this possible according to the rules or system?

(2) Be in two leagues as the same agent.
Is this impossible because OneShotAgent and SCML2020Agent are very different?

Please let me know so that I can decide on a future course of action.
If it is not possible, I will join one of them.

**Comment by yasserfarouk** [comment]
Both (1) and (2) are possible both technically and according to the rules.

OneShot agents that are marked to work in the Std/Collusion leagues will run through an adapter that we already provide. Nevertheless, they will not behave very well because some of the assumptions they will usually have will be broken. For example, delivery time is not negotiable in OneShot but is negotiable in Std/Collusion.

I suggest that you use option (1) but if you want you can just try option (2) first and see how does your agent behave. 

We do not officially support agents based on OneShotAgent is std/collusion.

**Reply by snippet2021** [reply]
I have an additional question about (1).

When I click “update” on the submit page, the active agent is overwritten, but if I press “Fork”, will two active agents be registered?

If so, can both active agents be edited at a later time？(update code or submit pdf)

**Comment by yasserfarouk** [comment]
Yes. When you press fork, a new agent is created. 

You can update either agent later but you can have only one active agent per login (we know it is a limitation but the system was originally designed for one agent in all tracks). 

The best way to participate in all tracks using option (1) is to register two teams (same members is ok) in the system one for Oneshot and another for std/collusion. You will need another email for that though.

**Reply by snippet2021** [reply]
>You can update either agent later but you can have only one active agent per login (we know it is a limitation but the system was originally designed for one agent in all tracks).

Does this mean that I can switch agents to edit by logging out and logging in? 
Also, the word of “active agent” refers to the editing target? And  both agents are eligible to participate in the league, right?

**Comment by yasserfarouk** [comment, marked as answer]
> Does this mean that I can switch agents to edit by logging out and logging in?

You can have multiple agents under the same user account. You can find all of them under the `submissions` tab. Only one of these will be active at any point of time and this is the one that participates in the online competition and the official competition. You can activate any agent by pressing the "Activate" button on its record in the `submissions` tab. 

The `Active Agent` tab shows the details of the currently active agent.
`Fork agent`, creates a new `submission` based on the current active agent.

To edit any submission, you have to activate it first as described above.

> Also, the word of “active agent” refers to the editing target? 

Yes the active agent is the one that is being edited under the `Active Agent` tab.


> And both agents are eligible to participate in the league, right?

You can have one active agent per user. To have two agents participating the competition, you need to have multiple users. 
For the online competition, it is OK to have multiple accounts with active agents in the same track. For  example, you can have two users both with a `one-shot` active agent.

For the official competition (at IJCAI), the team can have at most one agent per league. This means you cannot have two one-shot agents in the official competition (or std or collusion). You can though have one agent the one shot track using one user and another agent using a different user participating in std and collusion tracks (that is option (1) in your original post).

**Reply by snippet2021** [reply]
Thank you for the detailed explanation. I now understand how the submission process works.

So one person can have two accounts and submit agents of different leagues (oneshot and std/col), right?
I will try to create a new account with a different email address.

**Comment by yasserfarouk** [comment]
> So one person can have two accounts and submit agents of different leagues (oneshot and std/col), right?

Exactly. Thank you for your interest is SCML.
